lib/Stack.rb

```diff
@@ -4,21 +4,46 @@ def initialize
   end
 
   def push(element)
+    @store.push(element)
+    #@store.unshift(element)
   end
 
   def pop
+     #return @store.pop
+     last_element = @store.last
+     @store.delete(@store.last)
```

Comment:
I think this could cause problems if the last element appears multiple times in the stack, but I could be wrong.